splinterd/api/static/openapi.yml

```diff
@@ -327,6 +327,145 @@ paths:
         500:
           description: Internal service error
 
+  /user/notifications:
+    get:
+      description: Retrieve a list of all notifications
+      responses:
+        200:
+          description: Notifications have been retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                $ref: '#/components/schemas/Notification'
+        403:
+          description: User not authorized to access this endpoint
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+      description: Publish a list of notifications
+      responses:
+        200:
+          description: Notifications have been published
+        400:
+          description: Malformed notifications list
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        500:
+          description: Internal service error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  /user/users/{user_id}/notifications:
+    get:
+      description: Retrieve a list of notifications for a given user from the notifications database
+      parameters:
+        - name: user_id
+          in: path
+          description: user to retrieve notifications for
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Notifications have been retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                $ref: '#/components/schemas/Notification'
+        403:
```

Comment:
What determines of the user is authorized?

Reply:
If they A.) Have some admin permissions or B.) have the same user ID ad the requested resource. Not entirely sure on how that gets added to the request. Suggestions?

Reply:
Previously we talked about a bearer token

Reply:
So I guess "unauthorized" might be misleading in that description. 403 is forbidden. So a user could be authorized (i.e. logged in) but forbidden from accessing that resource (i.e have incorrect permissions).

Reply:
In any case, we should add the token documentation to this spec and apply it to the impacted routes.

Reply:
I don't believe we would put that in this spec. That would be part of the headers of the request, none of which are specified in any of the existing endpoints described in this file.